Yurii_Smazhnyi/3/PokerHands.py

```diff
@@ -0,0 +1,229 @@
+         "J", "Q", "K", "A")
+
+categories = ("Straight Flush", "Four of a kind",
+              "Full house", "Flush",
+              "Straight", "Three of a kind",
+              "Two pair", "One Pair",
+              "High card")
+
+
+def is_flush(hand):
+
+    """
+    Check is flush in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    for s in suits:
+
+        if hand.count(s) == 5:
+            
+            return True
+
+    return False
+
+
+def is_straight(hand):
+
+    """
+    Check is straight in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    card_in_row = 0
+
+    for c in cards:
+
+        if c in hand:
+
+            card_in_row += 1
+
+            if card_in_row == 5:
+
+                return True
+
+            continue
+        
+        card_in_row = 0      
+    
+    return False
+
+
+def is_straight_flush(hand):
+
+    """
+    Check is straight flush in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    if not is_flush(hand):
+
+        return False
+    
+    if not is_straight(hand):
+
+        return False
+    
+    return True
+
+
+def is_same_cards_in_hand(hand, count_of_cards):
+
+    """
+    Check is some count of cards in hand
+
+    @param hand: string hand with cards
+    @param count_of_cards: count of card that mast be same
+    @returns: bool result of check
+    """
+
+    for c in cards:
+
+        if hand.count(c) == count_of_cards:
+
+            return True
+
+    return False    
+
+
+def is_full_house(hand):
+
+    """
+    Check is full house in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    return is_same_cards_in_hand(hand, 2) and is_same_cards_in_hand(hand, 3)
+
+
+def is_four_of_a_kind(hand):
+
+    """
+    Check is flush in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    return is_same_cards_in_hand(hand, 4)
+
+
+def is_three_of_kind(hand):
+
+    """
+    Check is three of kind in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    return is_same_cards_in_hand(hand, 3)
+
+
+def is_two_pair(hand):
+
+    """
+    Check is two pair in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    pair_count = 0
+
+    for c in cards:
+
+        if hand.count(c) == 2:
+
+            pair_count += 1
+
+    return pair_count == 2    
+
+
+def is_one_pair(hand):
+
+    """
+    Check is pair in hand
+
+    @param hand: string hand with cards
+    @returns: bool result of check
+    """
+
+    return is_same_cards_in_hand(hand, 2)
+
+
+def get_rank_of_hand(hand):
+
+    """
+    Checks hand and returns it rank
+
+    @param hand: string hand with cards
+    @returns: int score of hand
+    """
+
+    if is_straight_flush(hand):
```

Comment:
Avoid multiple returns in function if it's possible. Replace it with variable and just return this variable in the end of the function.